File: kyw0716/level0/안전지대.js
function getIsBombNearBy(board, i, j) {
  for (let c = -1; c < 2; c++) {
    for (let r = -1; r < 2; r++) {
      if (c === 0 && r === 0) continue;
      if (
        i + c < 0 ||
        i + c > board.length - 1 ||
        j + r < 0 ||
        j + r > board.length - 1
      )
        continue;
      if (board[i + c][j + r] === 1) return true;
    }
  }
}

function solution(board) {
  var answer = 0;

  for (let i = 0; i < board.length; i++) {
    for (let j = 0; j < board[i].length; j++) {
      if (getIsBombNearBy(board, i, j) && board[i][j] !== 1) board[i][j] = 2;
    }
  }

  board.forEach((b) => {
    answer += b.filter((v) => v === 0).length;
  });

  return answer;
}

// https://school.programmers.co.kr/learn/courses/30/lessons/120866

File: sk000801/입문/안전지대.java
class Solution {
    public int solution(int[][] board) {
        int[][] pad = new int[board.length+2][board[0].length+2];
        int count = 0;
        for(int i = 0; i < pad.length; i++) {
            pad[0][i] = -1;
            pad[i][0] = -1;
            pad[board.length-1][i] = -1;
            pad[i][board.length-1] = -1;
        }
        for(int i = 1; i < pad.length-1; i++) {
            for(int j = 1; j < pad[i].length-1; j++) {
                if(board[i-1][j-1] == 1) {
                    pad[i][j+1] = 1;
                    pad[i][j-1] = 1;
                    pad[i][j] = 1;
                    pad[i+1][j] = 1;
                    pad[i+1][j+1] = 1;
                    pad[i+1][j-1] = 1;
                    pad[i-1][j+1] = 1;
                    pad[i-1][j] = 1;
                    pad[i-1][j-1] = 1;
                }
            }
        }
        for(int i = 1; i < pad.length-1; i++) {
            for(int j = 1; j < pad.length-1; j++) {
                if(pad[i][j] != 1) count++;
            }
        }
        return count;
    }
}